hypothesis-python/src/hypothesis/strategies.py

```diff
@@ -926,7 +927,32 @@ def text(
     elif isinstance(alphabet, SearchStrategy):
         char_strategy = alphabet
     else:
-        char_strategy = sampled_from(list(map(text_type, alphabet)))
+        if not isinstance(alphabet, Sequence):
+            note_deprecation(
+                'alphabet must be an ordered sequence, or tests may be '
+                'flaky and shrinking weaker, but a %r is not a type of '
+                'sequence.  This will be an error in future.'
+                % (type(alphabet),)
+            )
+        alphabet = list(alphabet)
+        non_string = [c for c in alphabet if not isinstance(c, string_types)]
+        print(non_string)
```

Comment:
Just remove this print statement, and we'll be ready to merge!

Reply:
done, actually I was quite busy these days and didn't have any clue for the error.
Thanks again for helping me out.